crates/cheatcodes/src/inspector.rs

```diff
@@ -1556,6 +1602,90 @@ impl Cheatcodes {
         }
     }
 
+    #[cold]
+    fn meter_gas_record<DB: DatabaseExt>(
+        &mut self,
+        interpreter: &mut Interpreter,
+        ecx: &mut EvmContext<DB>,
+    ) {
+        if matches!(interpreter.instruction_result, InstructionResult::Continue) {
+            self.gas_metering.gas_records.iter_mut().for_each(|record| {
+                if ecx.journaled_state.depth() == record.depth + 1 {
+                    println!(
+                        "[{:?}]: {:?} [{:?}] @ {:?}",
+                        revm::interpreter::OpCode::new(interpreter.current_opcode())
+                            .unwrap()
+                            .as_str(),
+                        interpreter.gas.spent(),
+                        record,
+                        ecx.journaled_state.depth(),
+                    );
+
+                    match interpreter.current_opcode() {
+                        op::CREATE |
+                        op::CALL |
+                        op::CALLCODE |
+                        op::DELEGATECALL |
+                        op::CREATE2 |
+                        op::STATICCALL |
+                        op::EXTSTATICCALL |
+                        op::EXTDELEGATECALL => {
+                            // Reset gas used when entering a new frame.
+                            self.gas_metering.last_gas_used = 0;
+
+                            match interpreter.current_opcode() {
+                                op::CREATE | op::CREATE2 => {
+                                    record.gas_used = record.gas_used.saturating_add(CREATE);
+                                }
+                                op::CALL | op::CALLCODE | op::DELEGATECALL => {
+                                    // See: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-150.md
+                                    let opcode_cost = if ecx
+                                        .journaled_state
+                                        .spec
+                                        .is_enabled_in(SpecId::TANGERINE)
+                                    {
+                                        700 + CALL_STIPEND
+                                    } else {
+                                        40 + CALL_STIPEND
+                                    };
+
+                                    record.gas_used = record.gas_used.saturating_add(opcode_cost);
+                                }
+                                op::EXTDELEGATECALL => {
+                                    record.gas_used =
+                                        record.gas_used.saturating_add(700 + CALL_STIPEND);
+                                }
+                                op::STATICCALL | op::EXTSTATICCALL => {
+                                    record.gas_used = record.gas_used.saturating_add(700);
+                                }
+                                _ => {}
+                            }
```

Comment:
Added some custom handling for these opcodes but wondering if you know a better approach as it doesn't take into account memory expansion, data size @klkvr